tools/osbuilder/rootfs-builder/agent-config.toml

```diff
@@ -0,0 +1,44 @@
+# Copyright 2022 Advanced Micro Devices, Inc.
```

Comment:
Could you either re-use, or refactor `docs/how-to/data/confidential-agent-config.toml.in` as it's pretty similar code?

Reply:
I think we should try to standardize the config at some point, but I'm not sure the one in the docs is up-to-date anymore (although it's not too different). It makes me a bit nervous to start changing the config right before the release. I think we might have some unnecessary entries in the new one.

Reply:
The agent config file should probably go under `kata-containers/tools/osbuilder/rootfs-builder`. This is where Alex has it currently. However, as Tobin mentioned, it's not standardized and it is tailored for the SEV use case:
`aa_kbc_params = "offline_sev_kbc::null"`

We could implement a variable and set it during the build. @stevenhorsman would that suffice?

 The docs config is being utilized by `.github/workflows/deploy-ccv0-demo.yaml`, and we can update it in this PR or wait for further documentation revisions.

We discussed the name of the agent config and decided to go with `/etc/agent-config.toml` as it's referenced in the tests repo:
`tests/integration/kubernetes/confidential/lib.sh`:    `local rootfs_agent_config="/etc/agent-config.toml"`

We could always update the name of the config in the tests repo if that is preferred.

@fitzthum Alex and I both had the same entries for the agent config in this PR and tests 4977. However, some may not be necessary. I'm pretty sure `docs/how-to/data/confidential-agent-config.toml.in` is out of date and missing required items. I could do some testing to determine what's actually required, but 

We should work on documentation updates before/during the freeze period. I can help.

Reply:
Hey, thanks for the discussion. I think we want to consolidate both agent-config.toml for the reason that if we have multiple it will be harder to keep them all update and in sync (I say this in shame as I'm adding another one to the tests repo at the moment, but I'll try and reconcile this), but if you want this to become the new 'main' version then I can see that it makes sense to have it in the `rootfs-builder` and the name is fine `agent-config.toml.in` makes sense to me. In the short term passing in the aa_kbc_params via and environment variable similar to how we do in https://github.com/kata-containers/tests/blob/926c91af076af51690df067e0d821cbfed82a571/integration/kubernetes/confidential/lib.sh#L156 sounds sensible and we might need to slightly expand the scope to include things like the http proxy.
If you want to spin out a separate issue for trying to update the doc reference to move to this one that's totally fine too, I just wanted to try and avoid us duplicating and diverging too much.
PS: I hope this rambling make sense, otherwise give me a shout!

Reply:
switched to a model with an agent-config.toml.in and using envsubst to template the kbc name

Reply:
Cool, thanks for this Alex. Post release we can have a proper discussion about what the officially 'sensitive' endpoints are and work to reconcile the agent config to remove duplication, but we are likely to be pretty close :)